app/src/main/java/com/framgia/fsalon/screen/bill/BillViewModel.java

```diff
@@ -0,0 +1,190 @@
+package com.framgia.fsalon.screen.bill;
+
+import android.app.Activity;
+import android.content.Context;
+import android.databinding.BaseObservable;
+import android.databinding.Bindable;
+import android.text.TextUtils;
+import android.widget.ArrayAdapter;
+
+import com.framgia.fsalon.BR;
+import com.framgia.fsalon.R;
+import com.framgia.fsalon.data.model.Bill;
+import com.framgia.fsalon.data.model.BillRequest;
+import com.framgia.fsalon.data.model.Service;
+import com.framgia.fsalon.data.model.Stylist;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Exposes the data to be used in the Bill screen.
+ */
+public class BillViewModel extends BaseObservable implements BillContract.ViewModel {
+    private BillContract.Presenter mPresenter;
+    private BillRequest mBillRequest;
+    private ArrayAdapter<Stylist> mStylistAdapter;
+    private ArrayAdapter<Service> mServiceAdapter;
+    private BillAdapter mAdapter;
+    private Stylist mStylist;
+    private Service mService;
+    private String mPrice;
+    private String mQty;
+    private Activity mActivity;
+    private Context mContext;
+
+    public BillViewModel(Activity activity) {
+        mActivity = activity;
+        mContext = activity.getApplicationContext();
+        mAdapter = new BillAdapter(mContext, new ArrayList<Bill>(), this);
+    }
+
+    @Override
+    public void onStart() {
+        mPresenter.onStart();
+    }
+
+    @Override
+    public void onStop() {
+        mPresenter.onStop();
+    }
+
+    @Override
+    public void setPresenter(BillContract.Presenter presenter) {
+        mPresenter = presenter;
+    }
+
+    @Override
+    public void onAddBillClick() {
+        if (mService == null || mStylist == null || TextUtils.isEmpty(mPrice) || TextUtils
+            .isEmpty(mQty)) {
+            return;
+        }
+        Bill bill = new Bill.Builder()
+            .setId(mService.getId())
+            .setStylistId(mStylist.getId())
+            .setPrice(Float.valueOf(mPrice))
+            .setQty(Integer.valueOf(mQty))
```

Comment:
ERROR: [PERFORMANCE][DM_BOXED_PRIMITIVE_FOR_PARSING] Boxing/unboxing to parse a primitive com.framgia.fsalon.screen.bill.BillViewModel.onAddBillClick()
http://findbugs.sourceforge.net/bugDescriptions.html#DM_BOXED_PRIMITIVE_FOR_PARSING